modules/afsocket/afsocket-source.c

```diff
@@ -847,8 +847,9 @@ _sd_open_stream(AFSocketSourceDriver *self)
     {
       /* NOTE: this assumes that fd 0 will never be used for listening fds,
        * main.c opens fd 0 so this assumption can hold */
+      gpointer fd = cfg_persist_config_fetch(cfg, afsocket_sd_format_listener_name(self));
       sock = GPOINTER_TO_UINT(
-               cfg_persist_config_fetch(cfg, afsocket_sd_format_listener_name(self))) -
+               fd) -
```

Comment:
This could be merged into a single line now that you've extracted the config_persist call.

`sock = GPOINTER_TO_UINT(fd) - 1;`

Reply:
Fixed 👍